### PR TITLE
driver/fakesensor: fix bugs.

### PR DESCRIPTION
**Pull request by buyuer**
1.fix crash casued by closing the handle during the reading.
2.fix set_interval fail sometime:When fakesensor was activated, csv file is not opened timely, so
the sensor->interval that set_interval() seted will be coverd by
interval form csv file.

Signed-off-by: buyuer <dingddding@163.com>

### CODE REVIEW

File: drivers/sensors/fakesensor.c [hunk 1]
```diff
@@ -49,7 +50,8 @@ struct fakesensor_s
   unsigned int batch;
   int raw_start;
   FAR const char *file_path;
-  sem_t run;
+  sem_t wakeup;
+  bool running;
```
Comment (hartmannathan) [inline drivers/sensors/fakesensor.c:54]: should this be:

`  volatile bool running;`

?

**Comment by acassis** [comment]
hi @buyuer why did you decide to close it? Did you manage to fix the issue other way?

**Comment by xiaoxiang781216** [comment]
@acassis here is the fix https://github.com/apache/incubator-nuttx/pull/4007

**Comment by acassis** [comment]
thank you @xiaoxiang781216

**Pull request closed without merging**